src/songuploader/utils.py

```diff
@@ -1,10 +1,24 @@
 from django.contrib.auth.mixins import LoginRequiredMixin
 from django.urls import reverse_lazy
 from django.views.generic import TemplateView, View
+from pytube import YouTube
+from songuploader.settings import MEDIA_ROOT
+from django.core.files import File
+from uploader.models import Submission
+import os 
 
 
 class LoginRequiredTemplateView(LoginRequiredMixin, TemplateView):
     login_url = reverse_lazy("login")
     redirect_field_name = "redirect_to"
 
     template_name: str = None
+
+def download_song(url: str, submission: Submission):
+    out_path = os.path.join("/tmp", f"{submission.user.username}_temp.wav");
+    YouTube(url=url).streams.first().download(output_path=os.path.dirname(out_path), filename=os.path.basename(out_path))
+    open_file = open(out_path, 'r')
+    song_file: File = File(open_file)
+    submission.song = song_file
```

Comment:
`submission.song` can be just the file path i think

Reply:
File path as string or another datatype. Python sadly doesnt say it..........

Reply:
https://docs.djangoproject.com/en/4.1/ref/models/fields/#django.db.models.FileField Here is saysy i have to open is and the provide it to a file in order to make a possible for the file field so to file field stuff. I have tried it with path etc. Thats where the codec problem origninates from